PR for SRVLOGIC-300, SRVLOGIC-190, SRVLOGIC-188, SRVLOGIC-192: Document "Serverless Logic" parent directory for Openshift Serverless Logic docs. (#76490)
openshift/openshift-docs: modules/serverless-logic-deploying-workflows-dev-mode.adoc

Author: Added

//
// * serverless/serverless-logic/serverless-logic-creating-managing-workflows.adoc

:_mod-docs-content-type: PROCEDURE
[id="serverless-logic-deploying-workflows-dev-mode_{context}"]
= Deploying workflows in Dev mode

You can deploy your local workflow on {ocp-product-title} in Dev mode. You can use this deployment to experiment and modify your workflow directly on the cluster, seeing changes almost immediately. Dev mode is designed for development and testing purposes. It is ideal for initial development stages and for testing new changes.

.Prerequisites

* You have {ServerlessLogicOperatorName} installed on your cluster.
* You have access to a {ServerlessLogicProductName} project with the appropriate roles and permissions to create applications and other workloads in {ocp-product-title}.
* You have installed the OpenShift CLI `(oc)`.

.Procedure

. Create the workflow configuration YAML file.
+
.Example `workflow-dev.yaml` file
[source,yaml]
----
apiVersion: sonataflow.org/v1alpha08
kind: SonataFlow
metadata:
  name: greeting <1>
  annotations:
    sonataflow.org/description: Greeting example on k8s!
    sonataflow.org/version: 0.0.1

>     sonataflow.org/profile: dev 

Reviewer: ```suggestion
    sonataflow.org/profile: dev <2>
```

spec:
  flow: 
    start: ChooseOnLanguage
    functions:
      - name: greetFunction
        type: custom
        operation: sysout
    states:
      - name: ChooseOnLanguage
        type: switch
        dataConditions:
          - condition: "${ .language == \"English\" }"
            transition: GreetInEnglish
          - condition: "${ .language == \"Spanish\" }"
            transition: GreetInSpanish
        defaultCondition: GreetInEnglish
      - name: GreetInEnglish
        type: inject
        data:
          greeting: "Hello from JSON Workflow, "
        transition: GreetPerson
      - name: GreetInSpanish
        type: inject
        data:
          greeting: "Saludos desde JSON Workflow, "
        transition: GreetPerson
      - name: GreetPerson
        type: operation
        actions:
          - name: greetAction
            functionRef:
              refName: greetFunction
              arguments:
                message:  ".greeting + .name"
        end: true
----
+
<1> is a workflow_name

> 

Reviewer: ```suggestion
<2> indicates that the workflow must be deployed in Dev mode
```

Author: Added

// Module included in the following assemblies:
//
// * serverless/serverless-logic/serverless-logic-creating-managing-workflows.adoc

:_content-type: PROCEDURE
[id="serverless-logic-deploying-workflows-dev-mode"]

> = Deploying workflows in Dev mode

Reviewer 2: Is "Dev mode" the official name or short for "Developer mode"?

Author: Yes, it is the official name for now.

> You can deploy your local workflow on OpenShift in Dev mode. This allows you to experiment and modify your workflow directly on the cluster, seeing changes almost immediately. Dev mode is designed for development and testing purposes. It is ideal for initial development stages and testing new changes.

Reviewer 2: Again, maybe use "OpenShift Container Platform", or ideally an attribute for it, instead of "OpenShift".

Also I would replace "This allows you to experiment..." with "You can use this deployment to experiment..."

And I would add a second "for" : "It is ideal for initial development stages and for testing new changes."

.Prerequisites

* {ServerlessLogicOperatorName} is installed on your cluster.

> * You have created a {ServerlessLogicProductName} project or have access to a project with the appropriate roles and permissions to create applications and other workloads in OpenShift Container Platform.

Reviewer 2: again, there might be an attribute for "OpenShift Container Platform". And if there isn't one maybe add it?

Reviewer 2: And I would change to just "You created"

Author: Added an attribute.

> * You have created your {ServerlessLogicProductName} project

Reviewer 2: ```suggestion
* You created your {ServerlessLogicProductName} project.
```

> * Install the OpenShift CLI `(oc)`.

Reviewer 2: ```suggestion
* You installed the OpenShift CLI `(oc)`.
```

.Procedure

> . Create the workflow configuration `YAML` file:

Reviewer 2: ```suggestion
. Create the workflow configuration YAML file.
```

kind: SonataFlow
metadata:
  name: greeting
  annotations:
    sonataflow.org/description: Greeting example on k8s!
    sonataflow.org/version: 0.0.1
    sonataflow.org/profile: dev 
spec:
  flow: 
    start: ChooseOnLanguage
    functions:
      - name: greetFunction
        type: custom
        operation: sysout
    states:
      - name: ChooseOnLanguage
        type: switch
        dataConditions:
          - condition: "${ .language == \"English\" }"
            transition: GreetInEnglish
          - condition: "${ .language == \"Spanish\" }"
            transition: GreetInSpanish
        defaultCondition: GreetInEnglish
      - name: GreetInEnglish
        type: inject
        data:
          greeting: "Hello from JSON Workflow, "
        transition: GreetPerson
      - name: GreetInSpanish
        type: inject
        data:
          greeting: "Saludos desde JSON Workflow, "
        transition: GreetPerson
      - name: GreetPerson
        type: operation
        actions:
          - name: greetAction
            functionRef:
              refName: greetFunction
              arguments:
                message:  ".greeting + .name"
        end: true
----

> . Navigate to the directory where the `YAML` file is contained, and deploy the application by applying the `YAML` file:

Reviewer 2: ```suggestion
. To deploy the application, applying the YAML file by entering the following command:
```

+
[source,terminal]
----
$ oc apply -f <filename> -n <your_namespace>
----
+
.Example command
[source,terminal]
----
$ oc apply -f workflow-dev.yaml
----

> . Verify the deployment and check the status of the the deployed workflow:

Reviewer 2: ```suggestion
. Verify the deployment and check the status of the deployed workflow by entering the following command::
```

+
[source,terminal]
----
$ oc get workflow -n <your_namespace> -w
----
+
.Example command
[source,terminal]
----

> $ oc get sonataflows

Reviewer 2: The example does not fit the generic syntax, there is no -n nor -w

Author: fixed it

----
+

> Ensure that the workflow `greeting` is listed and the status is `Running` or `Completed`.

Reviewer 2: ```suggestion
Ensure that your workflow, for example the `greeting` workflow, is listed and the status is `Running` or `Completed`.
```

Reviewer 2: You had generic instructions AND examples in the previous steps so I suggest modifying this one for uniformity

> . Edit or modify the workflow directly in the cluster: 

Reviewer 2: ```suggestion
. Edit or modify the workflow directly in the cluster by entering the following command: 
```

+
[source,terminal]
----
$ oc edit sonataflow <workflow_name> -n <your_namespace>
----
+
.Example command
[source,terminal]
----
$ oc edit sonataflow greeting -n <your_namespace>
----

> . After editing, save the changes. The Openshift Serverless Logic Operator will detect the changes and update the workflow accordingly.

Reviewer 2: I would just use the present tense here - "detects" and "updates" without "will"

> . Verify the status and logs of the workflow to ensure the changes are applied correctly: 

Reviewer 2: ```suggestion
. To ensure the changes are applied correctly, verify the status and logs of the workflow by entering the following commands: 
```

+
[source,terminal]
----
$ oc get sonataflows -n <your_namespace>
$ oc logs <workflow_pod_name> -n <your_namespace>

> ----

Reviewer 2: https://redhat-documentation.github.io/supplementary-style-guide/#commands-in-code-blocks  "Use a single command per code block for each procedure step. "

Reviewer 2: So please break this code block down into two code blocks, one for each command

Author: Done

> . Once you are done with testing, clean up the resources to avoid unnecessary usage:

Reviewer 2: ```suggestion
. After completing the testing, delete the resources to avoid unnecessary usage by running the following command:
```

// Module included in the following assemblies:
//
// * serverless/serverless-logic/serverless-logic-creating-managing-workflows.adoc

:_content-type: PROCEDURE
[id="serverless-logic-deploying-workflows-dev-mode_{context}"]
= Deploying workflows in Dev mode

You can deploy your local workflow on {ocp-product-title} in Dev mode. You can use this deployment to experiment and modify your workflow directly on the cluster, seeing changes almost immediately. Dev mode is designed for development and testing purposes. It is ideal for initial development stages and for testing new changes.

.Prerequisites

* {ServerlessLogicOperatorName} is installed on your cluster.
* You have created a {ServerlessLogicProductName} project or have access to a project with the appropriate roles and permissions to create applications and other workloads in {ocp-product-title}.

> * You have created your {ServerlessLogicProductName} project.

Author: Yes, correct! Removed

> * You have installed the OpenShift CLI `(oc)`.

Author: I am unable to see that attribute. 
![Uploading image.png…]()
